docs/erc1155-skills.md

```diff
@@ -0,0 +1,214 @@
+The AEA `erc1155_deploy` and `erc1155_client` skills demonstrate an interaction between two AEAs with the usage of a smart contract.
+
+* The `erc1155_deploy` skill deploys the smart contract, creates and mints items. 
+* The `erc1155_client` skill signs a transaction to complete a trustless trade with the counterparty.
+
+## Preparation instructions
+ 
+### Dependencies
+
+Follow the <a href="../quickstart/#preliminaries">Preliminaries</a> and <a href="../quickstart/#installation">Installation</a> sections from the AEA quick start.
+
+##Discussion
+
+The scope of the specific demo is to demonstrate how to deploy a smart contract and interact with it. For the specific use-case, we create two AEAs one that deploys and creates tokens inside the smart contract and the other that signs a transaction so we can complete an atomic swap. The smart contract we are using is an ERC1155 smart contract
+with a one-step atomic swap functionality. That means the trade between the two AEAs can be trustless.
+
+### Launch an OEF node
+In a separate terminal, launch a local OEF node (for search and discovery).
+``` bash
+python scripts/oef/launch.py -c ./scripts/oef/launch_config.json
+```
+
+Keep it running for all the following demos.
+
+## Demo: Ledger payment
+
+A demo to run a scenario with a true ledger transaction on Ethereum `ropsten` network. This demo assumes the client
+does not trust the deployer AEA to make the trade.
+
+### Create the deployer AEA (ledger version)
+
+Create the AEA that will deploy the contract.
+
+``` bash
+aea create my_erc1155_deploy
+cd my_erc1155_deploy
+aea add connection fetchai/oef:0.1.0
+aea add skill fetchai/erc1155_deploy:0.1.0
+aea add contract fetchai/erc1155:0.1.0
+aea install
+```
+Additionally, create the private key for the deployer AEA.
+
+Generate and add a key for Ethereum use:
+``` bash
+aea generate-key ethereum
+aea add-key ethereum eth_private_key.txt
+```
+
+
+### Create the client AEA (ledger version)
+
+In another terminal, create the AEA that will sign the transaction.
+
+``` bash
+aea create my_erc1155_client
+cd my_erc1155_client
+aea add connection fetchai/oef:0.1.0
+aea add skill fetchai/erc1155_client:0.1.0
+aea add contract fetchai/erc1155:0.1.0
+aea install
+```
+
+Additionally, create the private key for the client AEA.
```

Comment:
I don't think this step is needed? The keys will be generated automatically when running and this AEA does not make any transactions?